### ISSUE TITLE
Bug memory dataset: order on field(s) with nil value raise exception

### ISSUE THREAD

**Issue by wafcio** [rom-rb/rom#202]
I found bug in enumerable_dataset.rb (memory dataset) in sort_by (order) method. When one field is nil and you want call order on this column it will raise error.

Example: You have data: [{ name: 'Jane', age: 24 }, { name: 'Joe', age: nil }] and you want order(:age)

```
comparison of Array with Array failed
/.../rom-2874edfbe4f4/lib/rom/support/enumerable_dataset.rb:53:in `sort_by’
/.../rom-2874edfbe4f4/lib/rom/support/enumerable_dataset.rb:53:in `sort_by'
/.../rom-2874edfbe4f4/lib/rom/memory/dataset.rb:51:in `order'
```

**Comment by solnic** [comment]
Shall we put nils before numbers?

**Comment by wafcio** [comment]
based on csv exmaple:

**Comment by wafcio** [comment]
based on csv example:
id;name;age
1;Jane;24
2;Joe;

We have nil age for Joe, of course we can put for example 0 or " " but it doesn't look good

**Comment by solnic** [comment]
I'm just wondering what the order should be, should it be `nil, 0, 1` or `0, 1, nil` :) Some dbs allow you to configure that behavior.

**Comment by wafcio** [comment]
good point, but i think we can nil treat like 0

**Comment by cflipse** [comment]
not quite ... nil != 0, particularly if you include negative numbers in your sorting.

Nil values don't flow into the sort; the question is if they should lead or trail the sorted order.    Different RDBMS' have different opinions on if it goes first or last, so it's a judgement call, but they SHOULD NOT appear in the middle of the list.  Personally, I'd sort nil's to the end.

**Comment by solnic** [comment]
I agree with @cflipse - anybody up to fixing it? I'm planning to release 0.7.0 next Monday. I'm done with my tasks and won't have much time to do anything else.

**Comment by solnic** [comment]
Fixed via 74b74dc
